ci-operator/step-registry/openshift-logging/install-konflux-catalogsource/openshift-logging-install-konflux-catalogsource-commands.sh

```diff
@@ -0,0 +1,309 @@
+            echo "MCP updated successfully"
+            break
+        fi
+    done
+    if [[ ${updatedMachineCount} != "${machineCount}" ]]; then
+        run_command "oc get mcp,node"
+        run_command "oc get mcp worker -o yaml"
+        return 1
+    fi
+}
+
+function update_global_auth () {
+  # get the current global auth
+  run_command "oc extract secret/pull-secret -n openshift-config --confirm --to /tmp"; ret=$?
+  if [[ $ret -ne 0 ]]; then
+      echo "!!! fail to get the cluster global auth."
+      return 1
+  fi
+
+  # add quay.io/openshifttest auth to the global auth
+  new_dockerconfig="/tmp/new-dockerconfigjson"
+  openshifttest_auth_user=$(cat "/var/run/vault/mirror-registry/registry_quay_openshifttest.json" | jq -r '.user')
+  openshifttest_auth_password=$(cat "/var/run/vault/mirror-registry/registry_quay_openshifttest.json" | jq -r '.password')
+  openshifttest_registry_auth=`echo -n "${openshifttest_auth_user}:${openshifttest_auth_password}" | base64 -w 0`
+
+  stage_auth_user=$(cat "/var/run/vault/mirror-registry/registry_stage.json" | jq -r '.user')
+  stage_auth_password=$(cat "/var/run/vault/mirror-registry/registry_stage.json" | jq -r '.password')
+  stage_registry_auth=`echo -n "${stage_auth_user}:${stage_auth_password}" | base64 -w 0`
+
+  reg_brew_user=$(cat "/var/run/vault/mirror-registry/registry_brew.json" | jq -r '.user')
+  reg_brew_password=$(cat "/var/run/vault/mirror-registry/registry_brew.json" | jq -r '.password')
+  brew_registry_auth=`echo -n "${reg_brew_user}:${reg_brew_password}" | base64 -w 0`
+
+  jq --argjson a "{\"brew.registry.redhat.io\": {\"auth\": \"${brew_registry_auth}\"},\"quay.io/openshifttest\": {\"auth\": \"${openshifttest_registry_auth}\"},\"registry.stage.redhat.io\": {\"auth\": \"$stage_registry_auth\"}}" '.auths |= . + $a' "/tmp/.dockerconfigjson" > ${new_dockerconfig}
+
+ # run_command "cat ${new_dockerconfig} | jq"
+
+  # update global auth
+  ret=0
+  run_command "oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=${new_dockerconfig}" || ret=$?
+  if [[ $ret -eq 0 ]]; then
+      check_mcp_status
+      echo "update the cluster global auth successfully."
+  else
+      echo "!!! fail to add stage registry auth, retry and enable log..."
+      sleep 1
+      ret=0
+      run_command "oc --loglevel=10 set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=${new_dockerconfig}" || ret=$?
+      if [[ $ret -eq 0 ]]; then
+        echo "update the cluster global auth successfully after retry."
+      else
+        echo "!!! still fail to add stage registry auth after retry"
+        return 1
+      fi
+  fi
+}
+
+# create ICSP for connected env.
+function create_icsp_connected () {
+
+    run_command "oc delete imagecontentsourcepolicies.operator.openshift.io/brew-registry --ignore-not-found=true"
+
+    if [[ -z ${MULTISTAGE_PARAM_OVERRIDE_LOGGING_TEST_VERSION} ]] ; then
```

Comment:
if [[   ${MULTISTAGE_PARAM_OVERRIDE_LOGGING_TEST_VERSION}  ==~ "registry-proxy.engineering.redhat.com/rh-osbs/iib" ]] ;then
  use 'registry.stage.redhat.io as proxy.
elif [[   ${MULTISTAGE_PARAM_OVERRIDE_LOGGING_TEST_VERSION}  ==~ "registry-proxy.engineering.redhat.com/rh-osbs/iib" ]]; then
   use quay.io/redhat-user-workloads as proxy registry.
else
    echo "unsupport catalogsource images ${MULTISTAGE_PARAM_OVERRIDE_LOGGING_TEST_VERSION}
    exit 1
fi

Reply:
I've thought that way, but when creating the IDMS, we have to get the test logging version.